modules/web-console/frontend/app/components/page-configure/components/modal-import-models/component.js

```diff
@@ -1006,63 +1006,85 @@ export class ModalImportModels {
-                                    label: drv.jdbcDriverJar,
-                                    value: {
-                                        jdbcDriverJar: drv.jdbcDriverJar,
-                                        jdbcDriverClass: drv.jdbcDriverCls
-                                    }
+                                _.forEach(drivers, (drv) => {
+                                    $scope.jdbcDriverJars.push({
+                                        label: drv.jdbcDriverJar,
+                                        value: {
+                                            jdbcDriverJar: drv.jdbcDriverJar,
+                                            jdbcDriverClass: drv.jdbcDriverCls
+                                        }
+                                    });
                                 });
-                            });
-
-                            $scope.ui.selectedJdbcDriverJar = $scope.jdbcDriverJars[0].value;
-
-                            $scope.importDomain.action = 'connect';
-                            $scope.importDomain.tables = [];
-                            this.selectedTables = [];
-                        }
-                        else {
-                            $scope.importDomain.jdbcDriversNotFound = true;
-                            $scope.importDomain.button = 'Cancel';
-                        }
-                    })
-                    .then(() => {
-                        $scope.importDomain.info = INFO_CONNECT_TO_DB;
 
-                        Loading.finish('importDomainFromDb');
-                    });
-            });
+                                $scope.ui.selectedJdbcDriverJar = $scope.jdbcDriverJars[0].value;
+
+                                $scope.importDomain.action = 'connect';
+                                $scope.importDomain.tables = [];
+                                this.selectedTables = [];
+                            }
+                            else {
+                                $scope.importDomain.jdbcDriversNotFound = true;
+                                $scope.importDomain.button = 'Cancel';
+                            }
+                        })
+                        .then(() => {
+                            $scope.importDomain.info = INFO_CONNECT_TO_DB;
+
+                            Loading.finish('importDomainFromDb');
+                        });
+                });
+        }
+
+        this.agentIsAvailable$ = this.agentMgr.connectionSbj.pipe(
+            pluck('state'),
+            map((state) => state !== 'AGENT_DISCONNECTED')
+        );
+
+        this.domainData$ = this.agentIsAvailable$.pipe(
+            switchMap((agentIsAvailable) => {
+                if (!agentIsAvailable)
+                    return of(EMPTY);
+
+                return from(fetchDomainData());
+            })
+        );
+
+        this.subscribers$ = merge(
+            this.subscription,
+            this.agentIsAvailable$,
```

Comment:
No need to subscribe to `agentIsAvailable$`, you've already subscribed to other observables (`domainData$`, `agentIsAvailable$` in template).

Reply:
Yes. Removed subsricption.